apps/web/src/views/Voting/components/CastVoteModal/MainView.tsx

```diff
@@ -35,6 +38,102 @@ interface MainViewProps {
   onDismiss: CastVoteModalProps['onDismiss']
 }
 
+type VeMainViewProps = {
+  vote?: {
+    label: string
+    value: number
+  }
+  isLoading?: boolean
+  isPending?: boolean
+  isError?: boolean
+  total: number
+  disabled?: boolean
+  veCakeBalance?: number
+  onConfirm?: () => void
+  onDismiss?: CastVoteModalProps['onDismiss']
+  block: number
+}
+
+export const VeMainView = ({
+  vote,
+  total,
+  isPending,
+  isLoading,
+  isError,
+  onConfirm,
+  onDismiss,
+  disabled,
+  block,
+  veCakeBalance,
+}: VeMainViewProps) => {
+  const { t } = useTranslation()
+
+  return (
+    <>
+      <ModalInner>
+        {vote ? (
+          <>
+            <Text color="secondary" mb="8px" textTransform="uppercase" fontSize="12px" bold>
+              {t('Voting For')}
+            </Text>
+            <TextEllipsis bold fontSize="20px" mb="8px" title={vote.label}>
+              {vote.label}
+            </TextEllipsis>
+          </>
+        ) : null}
+
+        <Text color="secondary" textTransform="uppercase" bold fontSize="14px">
+          {t('Your voting power at block')}
+          <StyledScanLink useBscCoinFallback href={getBlockExploreLink(block, 'block')} ml="8px">
```

Comment:
Better add multichain support in case needed in the future
```suggestion
          <StyledScanLink useBscCoinFallback href={getBlockExploreLink(block, 'block', chainId)} ml="8px">
```